components/root/root.tsx

```diff
@@ -222,7 +228,13 @@ export default class Root extends React.PureComponent<Props, State> {
         this.tabletMediaQuery = window.matchMedia(`(min-width: ${Constants.MOBILE_SCREEN_WIDTH + 1}px) and (max-width: ${Constants.TABLET_SCREEN_WIDTH}px)`);
         this.mobileMediaQuery = window.matchMedia(`(max-width: ${Constants.MOBILE_SCREEN_WIDTH}px)`);
 
+        this.smallSidebarMediaQuery = window.matchMedia(`(max-width: ${Constants.SMALL_SIDEBAR_BREAKPOINT}px)`);
```

Comment:
To avoid cluttering up the Root component with more stuff, would you mind pulling this out into a separate component that's just responsible for updating this state while rendering nothing? We probably should've done the same for the code that watches for when the window is resized, but that's existing code so need to change that now.

You could then add that component to the `render` method below inside the `CompassThemeProvider` that also contains the `SidebarRight` so that it only runs while one of the sidebars is visible. The LHS is part of a child component of that, so we just need to make sure that the RHS exists to track that

Reply:
`MediaQuery` related to `Sidebarbar` has been moved. Should we move the existing `MediaQuery` as well?

Reply:
That's okay where it is. This PR's already rather large, so we don't need to do it here. We're also planning on doing some cleanup which will move logic out of the `Root` component in the next few months, so we can tackle it then